### PR TITLE
Give cookies a builder. Constrain dates to <= Dec 31, 9999.

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/Cookie.java [hunk 1]
```diff
@@ -85,7 +107,7 @@ public boolean persistent() {
 
   /**
    * Returns the time that this cookie expires, in the same format as {@link
-   * System#currentTimeMillis()}. This is {@link Long#MAX_VALUE} if the cookie is not persistent, in
+   * System#currentTimeMillis()}. This is December 31, 9999 if the cookie is not persistent, in
```
Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/Cookie.java]: How is calling code supposed to check against this value? Shouldn't the constant be on this type?

File: okhttp/src/main/java/okhttp3/Cookie.java [hunk 2]
```diff
@@ -403,7 +427,120 @@ private static String parseDomain(String s) {
...
+
+    public Builder expiresAt(long expiresAt) {
+      if (expiresAt <= 0) expiresAt = Long.MIN_VALUE;
+      if (expiresAt > HttpDate.MAX_DATE) expiresAt = HttpDate.MAX_DATE;
+      this.expiresAt = expiresAt;
+      this.persistent = true;
+      return this;
+    }
+
+    public Builder domain(String domain, boolean hostOnly) {
```
Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/Cookie.java]: Worth providing two overloads instead of a boolean?

Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/Cookie.java]: I was definitely tempted. Can do.

File: okhttp/src/main/java/okhttp3/Cookie.java (hunk 1 above)
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/Cookie.java]: The calling code should just check persistent or not.

Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/Cookie.java]: (ie. `persistent()`)

Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/Cookie.java]: Oh, got it.

**Comment by JakeWharton** [comment]
LGTM

**Pull request merged**